common/src/main/java/com/lightstep/tracer/shared/AbstractTracer.java

```diff
@@ -397,78 +389,63 @@ private boolean sendReportWorker(boolean explicitRequest) {
-                    error("Exception creating Thrift client. Disabling tracer.", e);
-                    disable();
-                    return false;
-                }
-            }
         }
 
-        ReportRequest req = new ReportRequest();
-        req.setRuntime(runtime);
-        req.setSpan_records(spans);
-        req.setTimestamp_offset_micros(clockState.offsetMicros());
+        ReportRequest.Builder reqBuilder = ReportRequest.newBuilder();
+        reqBuilder.setReporter(reporter);
+        reqBuilder.setAuth(auth);
+        for (Span span : spans) {
+            reqBuilder.addSpans(span);
+        }
+        // TODO casting from long to int?
+        reqBuilder.setTimestampOffsetMicros((int) clockState.offsetMicros());
 
         if (clientMetrics != null) {
-            req.setInternal_metrics(clientMetrics.toThrift());
+            reqBuilder.setInternalMetrics(clientMetrics.toGrpc());
         }
 
         try {
             long originMicros = nowMicrosApproximate();
             long originRelativeNanos = System.nanoTime();
-            ReportResponse resp = client.Report(auth, req);
+            ReportResponse resp = client.report(reqBuilder.build());
 
-            if (resp.isSetTiming()) {
+            if (resp.hasReceiveTimestamp() && resp.hasTransmitTimestamp()) {
                 long deltaMicros = (System.nanoTime() - originRelativeNanos) / 1000;
                 long destinationMicros = originMicros + deltaMicros;
                 clockState.addSample(originMicros,
-                        resp.getTiming().getReceive_micros(),
-                        resp.getTiming().getTransmit_micros(),
-                        destinationMicros);
+                    Util.protoTimeToEpochMicros(resp.getReceiveTimestamp()),
+                    Util.protoTimeToEpochMicros(resp.getTransmitTimestamp()), destinationMicros);
             } else {
                 warn("Collector response did not include timing info");
             }
 
             // Check whether or not to disable the tracer
-            if (resp.isSetCommands()) {
-                for (Command command : resp.commands) {
-                    if (command.disable) {
+            if (resp.getCommandsCount() != 0) {
+                for (Command command : resp.getCommandsList()) {
+                    if (command.getDisable()) {
                         disable();
                     }
                 }
             }
 
+            // TODO probably should check errors field of the response here?
+
             debug(String.format("Report sent successfully (%d spans)", spans.size()));
             return true;
-
-        } catch (TApplicationException e) {
-            // Log as this probably indicates malformed spans
-            error("TApplicationException: error from collector", e);
-            return false;
-        } catch (TException x) {
+        } catch (Exception e) {
+            // TODO Catch more specific grpc exceptions
```

Comment:
Yes, good point, I haven't gotten around to figuring out the specific exceptions thrown by GRPC client - will catch/handle those + a final Exception to avoid crashing customer application.